base-lite/ub/ub.go

```diff
@@ -0,0 +1,313 @@
+package main
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"regexp"
+	"sort"
+	"strings"
+	"text/template"
+
+	"golang.org/x/sys/unix"
+)
+
+func ensure(envVar string) bool {
+	_, found := os.LookupEnv(envVar)
+	return found
+}
+
+func path(filePath string, operation string) bool {
+	switch operation {
+
+	case "readable":
+		return unix.Access(filePath, unix.R_OK) == nil
+	case "executable":
+		info, err := os.Stat(filePath)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "Error checking executable status of file %s: %s", filePath, err)
```

Comment:
nit: Use `%q` instead of `%s` for user/externally provided strings in error messages to ensure they are escaped/have clear boundaries